### DISCUSSION TITLE
Can a transaction be used to insert into one table and update another table?

### DISCUSSION THREAD

**Discussion by merrydance** [prisma/prisma#19436] [Q&A] [answered]
I am developing a small order execution system. I have tables for warehouses, materials, inbound orders, and inventory. When a record is inserted into the inbound order table, I need to update the inventory table. The inventory quantity should be increased by the quantity in the inbound order table, the inventory value should be increased by the total amount in the inbound order table, and the average cost in the inventory table should be the total inventory value divided by the inventory quantity. Do I need to use a transaction to handle this? I didn't see a dedicated function for nested updates.

**Comment by ludralph** [comment, marked as answer]
Hi @merrydance 👋 

Yes, using a transaction is recommended to handle the scenario you described in Prisma ORM. Transactions ensure that a group of database operations is executed atomically, meaning either all the operations succeed or none of them take effect. Based on the information you provided, you can choose either the sequential operations approach or the interactive transactions approach, depending on your specific requirements and preferences. In your case, since you need to update the inventory table when inserting a record into the inbound order table, the [interactive transaction](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions) approach seems more appropriate. You can encapsulate the necessary Prisma Client queries inside the transaction function to ensure they are executed atomically.
for example:
```js
import { PrismaClient, Prisma } from '@prisma/client';

const prisma = new PrismaClient();

async function createInboundOrder(orderData) {
  const { quantity, amount } = orderData;

  try {
    await prisma.$transaction(async (tx) => {
      // Create the inbound order record
      const inboundOrder = await tx.inboundOrder.create({ data: orderData });

      // Update the inventory table
      await tx.inventory.update({
        where: { warehouseId: orderData.warehouseId, materialId: orderData.materialId },
        data: {
          quantity: {
            increment: quantity
          },
          value: {
            increment: amount
          },
          averageCost: {
            set: amount / quantity
          }
        }
      });
    });

    console.log('Inbound order created successfully.');
  } catch (error) {
    console.error('Error creating inbound order:', error);
  }
}

// Example usage
const orderData = {
  warehouseId: 1,
  materialId: 2,
  quantity: 10,
  amount: 100
};

createInboundOrder(orderData)
  .finally(() => {
    prisma.$disconnect();
  });
```
This example uses the $transaction API with an async function. The provided function receives an instance of the Prisma Client (here named tx), allowing you to perform Prisma queries within the transaction. If any error occurs during the transaction, it will be caught in the catch block, ensuring that the changes made to the database are rolled back.

**Reply by merrydance** [reply]
Thank you for your answer, I appreciate it very much.
